### PR TITLE
Migrating to new PM

### PR DESCRIPTION
**Pull request by neildhickey**
Initial preparation commit for migrating to using the new LLVM pass manager

### CODE REVIEW

File: include/LLVMSPIRVLib.h [hunk 1]
```diff
@@ -50,19 +50,19 @@ namespace llvm {
...
-void initializeSPIRVLowerConstExprPass(PassRegistry &);
-void initializeSPIRVLowerSPIRBlocksPass(PassRegistry &);
-void initializeSPIRVLowerOCLBlocksPass(PassRegistry &);
-void initializeSPIRVLowerMemmovePass(PassRegistry &);
-void initializeSPIRVLowerSaddWithOverflowPass(PassRegistry &);
-void initializeSPIRVRegularizeLLVMPass(PassRegistry &);
-void initializeSPIRVToOCL12Pass(PassRegistry &);
-void initializeSPIRVToOCL20Pass(PassRegistry &);
-void initializePreprocessMetadataPass(PassRegistry &);
+void initializeLLVMToSPIRVLegacyPass(PassRegistry &);
```
Comment (AlexeySachkov) [inline include/LLVMSPIRVLib.h:53]: I was always wondering why do we have these declarations in `LLVMSPIRVLib.h`? Do we really provide those as part of our public interface?

But this is for sure not a problem of this PR, just a question/observation.

**Review by AlexeySachkov — APPROVED** [review]

**Comment by AlexeySachkov** [comment]
@neildhickey, @svenvh, any plans for this PR? I don't have any objections, so I suggest that we rebase it, wait for testing results and submit it

**Comment by neildhickey** [comment]
I'm trying to rebase and get it working again so we can merge

**Review by AlexeySachkov — APPROVED** [review]

**Review by svenvh — APPROVED** [review]

**Comment by Nuullll** [comment]
Ah, it seems this PR implicitly "revert"s #904 
May I know the reason or should I create another PR to expose the `SPIRVToOCL12`, `SPIRVToOCL20` in the header again?

**Comment by neildhickey** [comment]
Oh, apologies, that wasn't really my intention, I guess because I moved it to a base class this was the side effect. Feel free to expose it again if that is what is needed. I plan to migrate to using the new PM soon, but there was an issue with pass order assumption that I need to fix, but I'll make sure to maintain the interface exposed when I do that.

**Comment by Nuullll** [comment]
> Oh, apologies, that wasn't really my intention, I guess because I moved it to a base class this was the side effect. Feel free to expose it again if that is what is needed. I plan to migrate to using the new PM soon, but there was an issue with pass order assumption that I need to fix, but I'll make sure to maintain the interface exposed when I do that.

Thanks! I've created another PR #1023

**Pull request merged**